src/fee-calculator.ts

```diff
@@ -26,34 +37,84 @@ export class FeeCalculator {
     );
     if (!balanceData) return;
 
-    // Get first date from balance data
-    const balanceDates = Object.keys(balanceData.balances).sort();
-    if (balanceDates.length === 0) return;
+    // Get all dates from balance data and sort chronologically
+    const sortedDates = Object.keys(balanceData.balances).sort();
+    if (sortedDates.length === 0) return;
 
-    const firstDate = balanceDates[0]!;
-    const firstBalance = balanceData.balances[firstDate]!;
+    // Process all dates to create daily entries
+    const dailyEntries = this.createDailyEntries(
+      sortedDates,
+      balanceData.balances,
+    );
 
-    // Create fee report
+    // Create and write fee report
     const feeReport: FeeReport = {
       metadata: {
         chain_id: CHAIN_IDS.ARBITRUM_NOVA,
       },
       distributors: {
-        [firstDistributorAddress]: [
-          {
-            date: firstDate,
-            start_balance_wei: firstBalance.balance_wei,
-            end_balance_wei: firstBalance.balance_wei,
-            balance_change_wei: FIRST_DAY_BALANCE_CHANGE,
-            distributions_wei: NO_DISTRIBUTIONS,
-            distributions_count: NO_DISTRIBUTIONS_COUNT,
-            total_wei: FIRST_DAY_BALANCE_CHANGE, // balance_change + distributions = 0 + 0
-          },
-        ],
+        [firstDistributorAddress]: dailyEntries,
       },
     };
 
-    // Write the report
     this.fileManager.writeFeeReport(feeReport);
   }
+
+  private createDailyEntries(
+    sortedDates: string[],
+    balances: { [date: string]: { block_number: number; balance_wei: string } },
+  ): FeeReportEntry[] {
+    const dailyEntries: FeeReportEntry[] = [];
+    let previousBalanceWei: string | null = null;
+
+    for (const date of sortedDates) {
+      const currentBalance = balances[date]!;
+      const balanceChangeWei = this.calculateBalanceChange(
+        currentBalance.balance_wei,
+        previousBalanceWei,
+      );
+
+      dailyEntries.push(
+        this.createDailyEntry(
+          date,
+          currentBalance.balance_wei,
+          balanceChangeWei,
+        ),
+      );
+      previousBalanceWei = currentBalance.balance_wei;
+    }
+
+    return dailyEntries;
+  }
+
+  private calculateBalanceChange(
+    currentBalanceWei: string,
+    previousBalanceWei: string | null,
+  ): string {
+    if (previousBalanceWei === null) {
+      return FIRST_DAY_BALANCE_CHANGE;
```

Comment:
Shouldnt the first day balance change just be the balance at the end of that day? we could achieve this by just passing in an initial 0 previousbalance?